doc/interop-test-descriptions.md

```diff
@@ -1011,16 +1011,49 @@ languages. Therefore they are not part of our interop matrix.
 #### rpc_soak
 
 The client performs many large_unary RPCs in sequence over the same channel.
-The number of RPCs is configured by the experimental flag, `soak_iterations`.
+
+This test is configurable via a few different command line flags:
+
+* `soak_iterations`: controls the number of RPCs to perform.
+
+* `soak_max_failures`: an inclusive upper limit on the number of RPC failures
+  that should be tolerated (i.e. after which the test process should
+  still exit 0). A failure is considered to be either a non-OK status or an RPC
+  whose latency exceeds `soak_per_iteration_max_acceptable_latency_ms`.
+
+* `soak_per_iteration_max_acceptable_latency_ms`: an inclusive upper limit
+  on the latency of a single RPC in order for that RPC to be considered
+  successful.
+
+* `soak_overall_timeout_seconds`: the overall number of seconds after which
+  the test should stop and fail if `soak_iterations` have not yet been
+  completed.
+
+The following suggestions are optional but encouraged to improve debuggability:
+
+* Implementations should avoid setting RPC deadlines and should instead
+  wait for each RPC to complete.
+
+* Implementations should log the number of milliseconds that each RPC takes.
+  Additionally, implementations should use a histogram of RPC latencies
+  to log interesting latency percentiles at the end of the test (e.g. median,
+  90th, and max latency percentiles).
+
+* Implementations should perform all RPCs first and then examine the number
+  of failed RPCs in a second pass. Mainly, implementations should not quit early
+  after exceeding `soak_max_failures`, since the total number of failures can
+  be interesting debug data.
 
 #### channel_soak
 
-The client performs many large_unary RPCs in sequence. Before each RPC, it
-tears down and rebuilds the channel. The number of RPCs is configured by
-the experimental flag, `soak_iterations`.
+Similar to rpc_soak, but this time each RPC is performed on a new channel. The
+channel is created just before each RPC and is destroyed just after.
```

Comment:
In Java we can wait for the channel to actually terminate. But I think C and Go don't have that ability (C because of subchannel sharing and Go 'just cause'). It'd be helpful to say that the channel shutdown time need not be included in the latency measurement. I wonder if it'd be better to not include channel destruction time at all, because it isn't actually a standardized concept, but I could go either way with that.

Reply:
Good point. I've updated to explicitly not include channel teardown in the latency measurement, since what it entails for a user to tear down a channel differs a lot between languages